lerobot/common/policies/diffusion/modeling_diffusion.py

```diff
@@ -749,3 +768,264 @@ def forward(self, x: Tensor, cond: Tensor) -> Tensor:
         out = self.conv2(out)
         out = out + self.residual_conv(x)
         return out
+
+
+class TransformerForDiffusion(nn.Module):
+    def __init__(self, config: DiffusionConfig, cond_dim: int):
+        super().__init__()
+        self.config = config
+
+        # compute number of tokens for main trunk and condition encoder
+        if config.n_obs_steps is None:
+            config.n_obs_steps = config.horizon
+
+        t = config.horizon
+        t_cond = 1
+        t_cond += config.n_obs_steps
+
+        input_dim = config.output_shapes["action"][0]
+        # input embedding stem
+        self.input_emb = nn.Linear(input_dim, config.diffusion_step_embed_dim)
+        self.pos_emb = nn.Parameter(torch.zeros(1, t, config.diffusion_step_embed_dim))
+        self.drop = nn.Dropout(config.p_drop_emb)
+
+        # cond encoder
+        self.time_emb = DiffusionSinusoidalPosEmb(config.diffusion_step_embed_dim)
```

Comment:
`diffusion_step_embed_dim` is found under the `# Unet` part of the config. Could you please consider making a `# Architecture shared params` section as well as `# Unet` and `# Transformer`? (or whatever else you think is suitable to make the configuration params clearer.